tests/unit/test_blob.py

```diff
@@ -3224,6 +3224,28 @@ def test_from_string_w_domain_name_bucket(self):
         self.assertEqual(blob.name, "b")
         self.assertEqual(blob.bucket.name, "buckets.example.com")
 
+    def test__lt__(self):
+        BUCKET = object()
+        blob_first = self._make_one("A", bucket=BUCKET)
+        blob_second = self._make_one("a", bucket=BUCKET)
+        blob_third = self._make_one("b", bucket=BUCKET)
+
+        self.assertLess(blob_first, blob_second)
+        self.assertLess(blob_second, blob_third)
+
+        self.assertEqual([blob_first, blob_second, blob_third], sorted([blob_third, blob_second, blob_first]))
+
+    def test__lt__w_other_instance(self):
+        with self.assertRaises(TypeError):
+            self._make_one("A", bucket=object()) < None
```

Comment:
The `TypeError` here won't be raised in Python 2.x.:

```python
Python 2.7.17 (default, Nov  7 2019, 10:07:09) 
[GCC 7.4.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo(object): pass
... 
>>> Foo() < None
False
>>> Foo() < object()
True
```

Reply:
Also, please move the construction of the blob instance outside the `with` statement, just in case *it* is the source of the `TypeError`:  we only want to trap the error from the comparison itself.  E.g.:

```python
    def test__lt__w_other_instance(self):
        blob = self._make_one("A", bucket=object())
        with self.assertRaises(TypeError):
             blob < None
```

Reply:
> The TypeError here won't be raised in Python 2.x.

That is interesting.  I could either _always_ manually raise a `TypeError` in said cases, or simply adjust the test to handle the specific Python 2.7 case. The former approach might be considered a backwards incompatible change in the event any python 2.7 people happen to be sorting said lists already. My preference would be for the latter case -- simply ignore that test in python 2.7.